scanner/indexer/indexer.go

```diff
@@ -57,15 +74,152 @@ func NewIndexer(ctx context.Context) (*Indexer, error) {
 
 	indexer, err := libindex.New(ctx, &opts, c)
 	if err != nil {
-		return nil, errors.Wrap(err, "creating libindex")
+		return nil, fmt.Errorf("creating libindex: %w", err)
 	}
 
-	return &Indexer{
+	return &indexerImpl{
 		indexer: indexer,
 	}, nil
 }
 
 // Close closes the indexer.
-func (i *Indexer) Close(ctx context.Context) error {
+func (i *indexerImpl) Close(ctx context.Context) error {
 	return i.indexer.Close(ctx)
 }
+
+// IndexContainerImage creates a ClairCore index report for a given container
+// image. The manifest is populated with layers from the image specified by a
+// URL. This method performs a partial content request on each layer to generate
+// the layer's URI and headers.
+func (i *indexerImpl) IndexContainerImage(
+	ctx context.Context,
+	manifestDigest claircore.Digest,
+	imageURL string,
+	opts ...Option,
+) (*claircore.IndexReport, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "scanner/backend/indexer")
+	o := makeOptions(opts...)
+	imgRef, err := parseContainerImageURL(imageURL)
+	if err != nil {
+		return nil, fmt.Errorf("parsing image URL: %w", err)
+	}
+	imgLayers, err := getContainerImageLayers(ctx, imgRef, o)
+	if err != nil {
+		return nil, fmt.Errorf("listing image layers (reference %q): %w", imgRef.String(), err)
+	}
+	httpClient := http.Client{Timeout: time.Minute}
+	manifest := &claircore.Manifest{
+		Hash: manifestDigest,
+	}
+	zlog.Info(ctx).
+		Str("image_reference", imgRef.String()).
+		Str("registry", imgRef.Context().RegistryStr()).
```

Comment:
this may not be necessary since it'll be part of the `image_reference`

Reply:
803ee475